subprojects/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/ConfigurationCacheGradlePropertiesIntegrationTest.groovy

```diff
@@ -179,15 +180,53 @@ class ConfigurationCacheGradlePropertiesIntegrationTest extends AbstractConfigur
         configurationCache.assertStateStored()
 
         when:
-        System.clearProperty('fromPropertiesFile')
+        System.clearProperty("$systemProp")
         configurationCacheRun fixture.task
 
         then:
-        outputDoesNotContain "because system property 'fromPropertiesFile' has changed"
+        outputDoesNotContain "because system property '$systemProp' has changed"
         outputContains fixture.expectedOutputBeforeChange
         configurationCache.assertStateLoaded()
 
         cleanup:
-        System.clearProperty('fromPropertiesFile')
+        System.clearProperty("$systemProp")
+    }
+
+    def "passing cli override invalidates cache entry only if property was read at configuration time"() {
+        given:
+        String systemProp = "fromPropertiesFile"
+        def configurationCache = newConfigurationCacheFixture()
+        def fixture = spec.createFixtureFor(this, systemProp)
+        fixture.setup()
+
+        when:
+        fixture.cleanPropertyRun()
+
+        then:
+        fixture.assertAfterFirstRun()
```

Comment:
IMO, having more fine-grained assertions in the fixture would make the test a bit more readable. For example, this could be expanded into:
```
configurationCache.assertStateStored()
fixture.assertSystemPropertyHadExpectedConfigurationTimeValue()
fixture.assertSystemPropertyHadExpectedExecutionTimeValue()
```
and for overriden run this would be:
```
configurationCache.assertStateStored()
fixture.assertConfigurationCacheInvalidatedBecauseOfPropertyChange()
fixture.assertSystemPropertyHadOverridenConfigurationTimeValue()
fixture.assertSystemPropertyHadOverridenExecutionTimeValue()
```

Reply:
Reasonable